Fakebook.Profile/Fakebook.Profile.DataAccess/ProfileRepository.cs

```diff
@@ -51,13 +72,52 @@ private static DomainProfile ToDomainProfile(EntityProfile profile)
         /// <exception type="ArgumentNullException">If the profile, the profile's email, or the profile's names are null,
         /// this will be thrown </exception>
         /// <returns>A representation of the profile as a DB Entity.</returns>
-        private static EntityProfile ToEntityProfile(DomainProfile profile)
+        private Task<EntityProfile> ToEntityProfile(DomainProfile profile)
         {
             if (profile == null || profile.Email == null || profile.FirstName == null || profile.LastName == null)
             {
                 throw new ArgumentNullException("Must have a domain profile, with an email and first and last name.");
```

Comment:
while we're here - this should be an ArgumentException, and referencing nameof(profile). ArgumentNullException is technically only for when one of the arguments themselves is null.

also - change the method naming - people expect that if there are two methods Abc and AbcAsync, that one is just the synchronous version of the other. and the one you're supposed to call and await should be the one with the Async suffix.

[local function](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions) would probably be a good fit for this case actually.

Reply:
Changed to ToEntityProfileAsync with a local function to handle async part

Reply:
I must not be implementing the solution correctly. I'm still getting a code smell from sonarcloud telling me to split the method up into two different ones where one handles exceptions and the other handles the rest.

Reply:
Fixed